src/services/Flow.ts

```diff
@@ -51,15 +51,15 @@ export class Flow {
     try {
       const flowData = await this.apiClient.get<FlowData>(`/flows/${this.id}`);
       this.version = flowData.version;
-      this.tags = flowData.tags;
+      this.tags = flowData.tags.map((tag) => `${tag.key}:${tag.value}`);
       this.description = flowData.description;
       this.steps = flowData.steps;
       this.name = flowData.name;
       this.status = flowData.status;
       this.created_at = flowData.created_at;
       this.updated_at = flowData.updated_at;
     } catch (err: any) {
-      throw new FlowError(`Failed to refresh flow with ID ${this.id}`, err);
+      throw new FlowError(err, `Failed to refresh flow with ID ${this.id}`, this.id);
```

Comment:
Maybe this catch should only wrap the apiClient call? Same in other places below.